### PR TITLE
CPLP-1440 rework db auditing

### PR DESCRIPTION
**Pull request by Phil91**
In this PR [CPLP-1440](https://jira.catena-x.net/browse/CPLP-1440) & [CPLP-1393](https://jira.catena-x.net/browse/CPLP-1393) is resolved.

Please also review the database [PR](https://github.com/tractusx-team-portal-onboarding/devenv/pull/136)

### CODE REVIEW

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Org.CatenaX.Ng.Portal.Backend.Framework.ErrorHandling;
+using Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.Auditing;
+
+namespace Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.AuditEntities;
+
+public static class EntityTypeBuilderExtensions
+{
+    public static string GetSql<TEntity>(this EntityTypeBuilder<TEntity> entity, TriggerEvent triggerEvent)
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs:32]: this method seems to be redundant (0 references, sql is now created by Laraue.EfCoreTriggers fluent-api in dbcontext )

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        if (!typeof(IAuditEntityV1).IsAssignableFrom(auditEntityType))
+        {
+            throw new ArgumentException($"Entity must derive from {nameof(IAuditEntityV1)}", nameof(auditEntityType));
+        }
+
+        _auditEntityType = auditEntityType;
+    }
+    
+    public string GetAuditSql<TEntity>(EntityTypeBuilder<TEntity> entity, TriggerEvent triggerEvent)
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs:55]: this method seems to be redundant (0 references, sql is now created by Laraue.EfCoreTriggers fluent-api in dbcontext )

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+               $"{snakeCaseStrategy.GetPropertyName(nameof(IAuditEntityV1.AuditV1OperationId), false)} ) " +
+               "SELECT " +
+               "gen_random_uuid(), " +
+               $"{string.Join(",", propertiesWithPrefix)}, " +
+               $"{_prefix}{entity.GetColumnName(nameof(IAuditEntityV1.AuditV1LastEditorId), tableName)}, " +
+               "CURRENT_DATE, " +
+               $"{(int)triggerEvent.GetOperationForTriggerEvent()} ;";
+    }
+
+    private static string? GetLastEditorIdColumnName<TEntity>(EntityTypeBuilder<TEntity> entity,
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs:89]: this method seems to be redundant (0 references, sql is now created by Laraue.EfCoreTriggers fluent-api in dbcontext )

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditLastEditor.cs [hunk 4]
```diff
@@ -18,23 +18,23 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
-using Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.Auditing;
-using Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.Entities;
-using Org.CatenaX.Ng.Portal.Backend.PortalBackend.PortalEntities.Enums;
+using Laraue.EfCoreTriggers.Common.TriggerBuilders;
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditLastEditor.cs:21]: remove redundant using-statements

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    /// <summary>
+    /// Get Column Name for the given property
+    /// </summary>
+    /// <param name="entity"></param>
+    /// <param name="propertyName"></param>
+    /// <param name="tableName"></param>
+    /// <typeparam name="TEntity"></typeparam>
+    /// <returns></returns>
+    public static string GetColumnName<TEntity>(this EntityTypeBuilder<TEntity> entity, string propertyName, string tableName)
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs:52]: this method seems to be redundant (0 references, sql is now created by Laraue.EfCoreTriggers fluent-api in dbcontext )
(Actually the whole class seems to be redundant)

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/TriggerEventExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+/// Extension methods for the trigger event
+/// </summary>
+public static class TriggerEventExtensions
+{
+    /// <summary>
+    /// Converts an <seealso cref="TriggerEvent"/> to an <seealso cref="AuditOperationId"/> 
+    /// </summary>
+    /// <param name="triggerEvent">The trigger event that should be converted</param>
+    /// <returns>Returns an <seealso cref="AuditOperationId"/></returns>
+    public static AuditOperationId GetOperationForTriggerEvent(this TriggerEvent triggerEvent) =>
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Auditing/TriggerEventExtensions.cs:36]: this method seems to be redundant (0 references, sql is now created by Laraue.EfCoreTriggers fluent-api in dbcontext )
(Actually the whole class seems to be redundant).

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs (hunk 1 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs]: removed

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs (hunk 2 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs]: removed

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs (hunk 3 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditEntityV1Attribute.cs]: removed

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditLastEditor.cs (hunk 4 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditLastEditor.cs]: removed

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs (hunk 5 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Auditing/EntityTypeBuilderExtensions.cs]: removed

File: src/portalbackend/PortalBackend.PortalEntities/Auditing/TriggerEventExtensions.cs (hunk 6 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Auditing/TriggerEventExtensions.cs]: removed

**Pull request merged**